benchmarks/test_stress_single_workflow_multiple_runs.py

```diff
@@ -0,0 +1,208 @@
+    )
+    plt.title("runtime + pending_time of each run_number")
+    plt.ylabel("seconds")
+    plt.legend()
+    plt.savefig(f"{folder_path}/time_of_each_run_number.png")
+
+
+def statistics_runtime_plot(folder_path: Path, stats: dict):
+    plt.clf()
+    plt.bar(stats["runtime"].keys(), stats["runtime"].values())
+    plt.title("runtime statistics")
+    plt.ylabel("seconds")
+    plt.savefig(f"{folder_path}/statistics_runtime.png")
+
+
+def statistics_count_plot(folder_path: Path, stats: dict):
+    plt.clf()
+    plt.pie(
+        stats["count"].values(),
+        labels=stats["count"].keys(),
+        autopct=lambda val: round(((val / 100) * sum(stats["count"].values()))),
+    )
+    plt.title("runs statistics")
+    plt.savefig(f"{folder_path}/statistics_runs_pie.png")
+
+
+def runs_time_histogram(folder_path: Path, df: pd.DataFrame):
+    plt.clf()
+    bin_size = 10
+    plt.hist(df["runtime"] + df["pending_time"], bin_size)
+
+    plt.xlabel(f"seconds (bin size = {bin_size})")
+    plt.ylabel("Number of runs")
+    plt.title("runtime + pending_time")
+    plt.grid(True)
+    plt.savefig(f"{folder_path}/runs_time_histogram.png")
+
+
+def create_plots(folder_path: Path, df: pd.DataFrame, stats: Dict) -> None:
+    logging.info("Creating plots...")
+    run_number_vs_runtime_plot(folder_path, df)
+    statistics_runtime_plot(folder_path, stats)
+    statistics_count_plot(folder_path, stats)
+    runs_time_histogram(folder_path, df)
+
+
+def main(workflow_name: str, number_of_submissions: int, to_analyze: bool):
+    results_folder_path = Path(f"benchmark_{workflow_name}")
+    original_results_path = Path(f"{results_folder_path}/original_results.csv")
+
+    if not to_analyze:
+        run_workflows(workflow_name, number_of_submissions)
+        results = get_results_when_completed(workflow_name)
+        try:
+            os.mkdir(results_folder_path)
+            logging.info("Saving original results...")
+        except FileExistsError:
+            logging.info("Folder already exists. Overwriting previous results...")
+
+        results.to_csv(original_results_path, index=False)
+    else:
+        results = pd.read_csv(original_results_path)
+        processed_results, workflow_stats = analyze(results)
+        create_plots(results_folder_path, processed_results, workflow_stats)
+
+        logging.info("Saving processed results...")
+        processed_results.to_csv(
+            f"{results_folder_path}/processed_results.csv", index=False
+        )
+
+        logging.info("Saving statistics...")
+        with open(f"{results_folder_path}/statistics.json", "w") as f:
+            json.dump(workflow_stats, f)
+    logging.info("Finished.")
+
+
+if __name__ == "__main__":
+    parser = ArgumentParser()
+    parser.add_argument(
+        "-w", "--workflow_name", type=str, help="Workflow name", required=True
+    )
+    parser.add_argument("-n", "--run_number", type=int, help="Number of runs to start")
```

Comment:
`-n, --number` so that we have nice matching of leading character in the short and long options